### ISSUE TITLE
Accessing HTTP request from session

### ISSUE THREAD

**Issue by tmbb** [aio-libs/sockjs#94]
SockJS sessions are initiated with a HTTP handshake.
Is there any way to access this request from the `Session` object?

It would be useful to be able to access the initial `request` object from `Session` to do things like identify and authenticate the client with the HTTP headers.

**Comment by tmbb** [comment]
(In the title I meant "from session" instead of "for session")

**Comment by ransomw** [comment]
> Is there any way to access this `request` from the Session object?

i was wondering about this too and wound up digging the source a little.. it's possible via "hook, extend, and subclass" patterns

```python
from sockjs import Session as SessionBase
from sockjs import SessionManager as SessionManagerBase

class MySession(SessionBase):

    def __init__(self, *args, **kwargs):
        super().__init__(*args, **kwargs)
        # check upstream bolts
        assert not hasattr(self, 'request')
        self.request = None


class MySessionManager(SessionManagerBase):
    """
    tacks the underlying aiohttp request onto the sockjs.Session
    """

    def __init__(self, *args, **kwargs):
        super().__init__('some_name', *args, **kwargs)
        self.factory = MySession

    def get(self, *args, **kwargs):
        session = super().get(*args, **kwargs)
        session.request = kwargs.get('request', None)
        return session


async def mySessionHandler(msg, sess):
    """
    msg (sockjs.protocol.SockjsMessage):
      a collections.namedtuple with fields 'tp', 'data',
      where 'tp' is one of the 'MSG_*' types exported by sockjs
    sess (sockjs.Session)
    """

    print("**** top of mySessionHandler")
    print(repr(sess.request))

    pass
```

which can be applied as

```python
mySessionManager = MySessionManager(
    aiowebApplication,
    mySessionHandler,
    aiowebApplication.loop,
    )
sockjs.add_endpoint(
    aiowebApplication, mySessionHandler,
    name=mySessionManager.name,
    prefix='/sockjs', # default
    manager=mySessionManager,
)
```

----

much thanks to the aio-libs team for all their great libraries!

**Comment by asvetlov** [comment]
I have no objections for publishing `.request` attribute but have not time for it.
The issue is waiting for the hero with prepared Pull Request

**Comment by spumer** [comment]
We need this feature! E.g for access to authentication headers! Make SockJS great again!

**Comment by asvetlov** [comment]
Fixed by #198
